src/clims/services/transition.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import absolute_import
+
+from django.db import transaction
+from clims.models.transition import Transition as TransitionModel
+from clims.models.transition import TransitionType
+from clims.models.location import SubstanceLocation as LocationModel
+from clims.models.container import Container as ContainerModel
+from clims.models.substance import Substance as SubstanceModel
+
+
+class TransitionService:
+    def __init__(self, app):
+        self._app = app
+
+    # TODO: CLIMS-401 - ensure atomic transaction only commits after plugin logic runs
+    # Atomic transaction:
+    # 1. create target location record
+    # 2. set "current" on source location record to false
+    # 3. create transition record
+    @transaction.atomic
+    def create(self, transition, work_batch_id):
+        """
+        Creates a new Transition
+        """
+        transition_type = transition["transition_type"]
```

Comment:
Previously, I wrote a comment about moving some of the code into the transition serializer. What I foremost reacted on is the use of strings to fetch variables, "transition_type", instead of using an ordinary object as input. I would like to have just that kind of logic focused at the serializer, where it's quite obvious what these strings would be. Here, these key-strings are spread out in a rather large chunk of code. Perhaps you have already addressed this? Then you can of course disregard this comment.

Reply:
Duly noted. I've committed some changes that get the worst offenders. The plan is to heavily refactor this later, so I'll let the nested objects slide for now. Thanks for all the feedback, Edvard, it really helped me a lot with understanding the architecture! Going to merge this beast now.